### PR TITLE
new data source `azurerm_network_manager_connectivity_configuration`

### LINKED ISSUE #25587: Support for azurerm_network_manager_connectivity_configuration data source
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

There is currently a Terraform resource for the `azurerm_network_manager_connectivity_configuration` but the AzureRM provider lacks a data source for the same type of resource.

I'm unsure if its not possible to get the Connectivity Configuration as a data source and perhaps that's why this hasn't been done - but I didn't see any Issues reporting the data source as missing so maybe it just hasn't been thought of yet.

### New or Affected Resource(s)/Data Source(s)

azurerm_network_manager_connectivity_configuration

### Potential Terraform Configuration

```hcl
data "azurerm_network_manager" "main" {
  resource_group_name = local.general_resource_group_name

  name = "${local.organization_name}-${local.general_resource_group_name}-nm"
}

data "azurerm_network_manager_connectivity_configuration" "config" {
  network_manager_id    = data.azurerm_network_manager.main.network_manager_id
  name = local.connectivity_group_name
}
```

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_manager_connectivity_configuration

### PR DESCRIPTION
**Pull request by teowa**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

new data source `azurerm_network_manager_connectivity_configuration`

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```hcl
=== RUN   TestAccNetworkManagerConnectivityConfigurationDataSource_basic
--- PASS: TestAccNetworkManagerConnectivityConfigurationDataSource_basic (204.44s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       204.459s
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* new data source `azurerm_network_manager_connectivity_configuration`

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #25587

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: website/docs/d/network_manager_connectivity_configuration.html.markdown [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+* `network_manager_id` - (Required) The ID of the Network Manager.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Network Manager Connectivity Configuration.
+
+* `applies_to_group` - A `applies_to_group` block as defined below.
```
Comment (jackofallops) [inline website/docs/d/network_manager_connectivity_configuration.html.markdown:40]:
```suggestion
* `applies_to_group` - An `applies_to_group` block as defined below.
```

File: website/docs/d/network_manager_connectivity_configuration.html.markdown [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+* `description` - The description of the Connectivity Configuration.
+
+* `global_mesh_enabled` - Whether global mesh is supported.
+
+* `hub` - A `hub` block as defined below.
+
+---
+
+A `applies_to_group` block exports the following:
```
Comment (jackofallops) [inline website/docs/d/network_manager_connectivity_configuration.html.markdown:54]:
```suggestion
An `applies_to_group` block exports the following:
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks @teowa - Just a couple things to take a look at below if you could.

Thanks!

**Review by katbyte — APPROVED** [review]
LGTM 🍵

**Pull request merged**